### ISSUE TITLE
Repeated loop in translation output

### ISSUE THREAD

**Issue by Pyran1** [LibreTranslate/LibreTranslate#183]
### Error in response
after doing request **_заранее_** in Russian language
I get enormous response.

Request:
```
const res = await fetch("https://libretranslate.com/translate", {
	method: "POST",
	body: JSON.stringify({
		q: "заранее",
		source: "ru",
		target: "en",
		format: "text"
	}),
	headers: { "Content-Type": "application/json" }
});

console.log(await res.json());
```

Response:
```
{
    "translatedText": "pre-pre- pre-pre- pre-pre- pre-pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre-"
}
```

**Comment by dingedi** [comment]
It's an [https://github.com/argosopentech/argos-translate](https://github.com/argosopentech/argos-translate) i think. 
it might be necessary to add a label or transfer the issues that concern argos-translate, so that it is more understandable ? @pierotofy @PJ-Finlay

**Comment by PJ-Finlay** [comment]
This issue of repeating the same output has been seen before. [CTranslate2](https://github.com/OpenNMT/CTranslate2) supports a repetition penalty which we've experimented with but haven't found a great solution.

@guillaumekln

**Comment by dingedi** [comment]
@PJ-Finlay  with [https://github.com/argosopentech/argos-translate/pull/234](https://github.com/argosopentech/argos-translate/pull/234)

before:
`./bin/argos-translate --from-lang=ru --to-lang=en "заранее" -> 
pre-pre- pre-pre- pre-pre- pre-pre- pre-pre- pre- pre- pre- pre-pre- pre- pre- pre-pre- pre- pre- pre- pre- pre-pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre- pre-`
after:
`./bin/argos-translate --from-lang=ru --to-lang=en "заранее" -> 
in advance pre-presiding priori, before early ahead`


the translation is much faster and the strange repetition problem is gone, the translation is however not good but that is another problem.
I think the repetition_penalty will solve a lot of things.

**Comment by PJ-Finlay** [comment]
Makes sense, I think it's worth having. 

I'm guessing the repitition problem is happening when the neural net is confused and doesn't have any good answer.

**Comment by guillaumekln** [comment]
@dingedi Did you check how it impacts other translations?

**Comment by dingedi** [comment]
I tried on a few random sentences and didn't see any change when the translations were already good, but there might be a better way to check it

**Comment by guillaumekln** [comment]
You should check how the BLEU score changes on a standard test. You can download test sets and score translations with [sacreBLEU](https://github.com/mjpost/sacreBLEU). For example:

```bash
# Get the source file:
sacrebleu -t wmt19 -l ru-en --echo src > input.ru

# Translate input.ru into output.en...

# Score the translation:
sacrebleu -i output.en -t wmt19 -l ru-en
```

**Comment by dingedi** [comment]
@guillaumekln @PJ-Finlay 
indeed, strange but the score is less high, add the repetition_penalty was maybe not a good idea.


for ru -> en

```bash
# WITHOUT repetition_penalty=2

{
 "name": "BLEU",
 "score": 26.7,
 "signature": "nrefs:1|case:mixed|eff:no|tok:13a|smooth:exp|version:2.0.0",
 "verbose_score": "60.6/33.4/20.9/13.6 (BP = 0.971 ratio = 0.971 hyp_len = 41408 ref_len = 42642)",
 "nrefs": "1",
 "case": "mixed",
 "eff": "no",
 "tok": "13a",
 "smooth": "exp",
 "version": "2.0.0"
}


# WITH repetition_penalty=2
{
 "name": "BLEU",
 "score": 19.0,
 "signature": "nrefs:1|case:mixed|eff:no|tok:13a|smooth:exp|version:2.0.0",
 "verbose_score": "57.3/27.0/15.2/8.8 (BP = 0.890 ratio = 0.895 hyp_len = 38179 ref_len = 42642)",
 "nrefs": "1",
 "case": "mixed",
 "eff": "no",
 "tok": "13a",
 "smooth": "exp",
 "version": "2.0.0"
}

```


launched with this bash file
```bash
#! /bin/bash

cat input.ru | while read LINE; do
    ./bin/argos-translate --from-lang=ru --to-lang=en "$LINE" >> output.en
done
```

**Comment by guillaumekln** [comment]
Yes, this is a big drop in score. The change should be reverted.

As mentioned in https://github.com/argosopentech/argos-translate/issues/212#issuecomment-968723509, the option probably cannot be set globally. The true fix is on the training side.

**Comment by PJ-Finlay** [comment]
Good catch, thanks.

@dingedi Do you mind contributing the BLEU script to Argos Translate `scripts/`?
